page_reader.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2016 Sebastian Berlin
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+
+import requests
+
+import hb_config
+
+pages = {}
+
+class PageReaderError(Exception): pass
+
+def read_json_value(page, name):
+    """Read the value from a page formatted as JSON, given a name."""
+
+    try:
+        page_content = _read_page(page)
+        page_json = json.loads(page_content)
+        return page_json[name]
+    except KeyError:
+        raise PageReaderError("Couldn't read variable '{}' from page '{}'"
+                              .format(name, page))
+
+def _read_page(page):
+    """Read the content of a page, given its title.
+
+    The content will be cached the first time a page is read, allowing
+    multiple calls to this function for the same page without sending
+    multiple requests.
+    """
+
+    if page in pages:
+        # Read cached page, if available.
+        page_content = pages[page]
+    else:
+        response = requests.post(
+            hb_config.apiurl,
+            data={
+                'action': "query",
+                'titles': page,
```

Comment:
Amazingly enough, character encoding doesn't seem to cause any trouble here. I did quick test and "User:FikarummetBot/Å" worked, both as `str` and `unicode`. I did changed the call to `get()´.